In [7]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service # 옵셥값 정의
from selenium.webdriver.common.by import By #~에 의한 명령
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())
options = Options()
options.add_argument("headless")
options.add_argument("window-size-1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")
# headless을 사용하려면 user-agent= 추가하기
driver = webdriver.Chrome(service = service, options = options) # service 문법 , 그래서 무조건 쓴다~, driver 특정사이트에 가서 크롤링해올수 있는 인스턴스객체

driver.get("https://davelee-fun.github.io/")

element = driver.find_element(By.TAG_NAME, "h1")
print(element.text)

time.sleep(3)


driver.quit()

Teddy의 선물 블로그


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 부모요소의 id속성값이 dev-course-list 인 자식요소class가 cousrse인 데이터만 출력해주세요.

In [25]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By

service = Service(ChromeDriverManager().install())
driver =webdriver.Chrome(service=service)

driver.get("https://crawlingtest.netlify.app/")
elements= driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")

for element in elements :
    print(element.text)

time.sleep(1)

driver.quit()

(초급) - 강사가 실제 사용하는 자동 프로그램 소개 [2]
(초급) - 필요한 프로그램 설치 시연 [5]
(초급) - 데이터를 엑셀 파일로 만들기 [9]
(초급) -     엑셀 파일 이쁘게! 이쁘게! [8]
(초급) -     나대신 주기적으로 파이썬 프로그램 실행하기 [7]
(초급) - 파이썬으로 슬랙(slack) 메신저에 글쓰기 [40]
(초급) - 웹사이트 변경사항 주기적으로 체크해서, 메신저로 알람주기 [12]
(초급) - 네이버 API 사용해서, 블로그에 글쓰기 [42]
(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 해당 사이트 내 커리큘럼 일정표 데이터 출력해주세요.

In [39]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
import time

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://crawlingtest.netlify.app/")
#elements = driver.find_elements(By.TAG_NAME,"td")
elements = driver.find_elements(By.CSS_SELECTOR,"table tr td")
for element in elements :
    print(element.text)

time.sleep(1)

driver.quit()

일정
커리큘럼 타이틀
난이도
5.1 ~ 6.15
나만의 엣지있는 블로그 사이트 만들기 (취미로 익히는 IT)
초급
6.16 ~ 7.31
파이썬과 데이터과학 첫걸음 (IT 기본기 익히기)
중급


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 아래 데이터만 추출해오세요!!
- (중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]

In [53]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
import time

service = Service(ChromeDriverManager().install())
driver= webdriver.Chrome(service=service)

driver.get("https://crawlingtest.netlify.app/")
elements = driver.find_elements(By.CSS_SELECTOR,"ul#dev_course_list > li.course")

print(elements[8].text)

time.sleep(1)
driver.quit

(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


<bound method ChromiumDriver.quit of <selenium.webdriver.chrome.webdriver.WebDriver (session="1f45e0be0262697c2771e4271fa04078")>>

#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news 인 값만 찾아오시오!

In [58]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time

service = Service(ChromeDriverManager().install())
driver=webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
element = driver.find_element(By.CSS_SELECTOR,"div.message > span.news")
print(element.text)
time.sleep(1)
driver.quit

뉴스: 머신러닝 강좌가 나왔어요~~~


<bound method ChromiumDriver.quit of <selenium.webdriver.chrome.webdriver.WebDriver (session="7047054a2d4b2f477745aa8dec9f9086")>>

#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news 인 값만 찾아오시오!
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기!

In [68]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from webdriver_manager.chrome import ChromeDriverManager
import time

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")

element01 = driver.find_element(By.NAME,"username")
# element01 = driver.find_element(By.ID,"username")
element01.clear()
element01.send_keys("error@error.com")

element02 = driver.find_element(By.NAME,"password")
# element02 = driver.find_element(By.ID,"password")
element02.clear()
element02.send_keys("1234")
#여기선 변수명 같이 사용해도 됨. 파이썬은 유연하니까~

time.sleep(3)

# element02.send_keys(Keys.RETURN) # 엔터키 입력
element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']") # CSS는 이렇게 쓰임, input 태그 많아요ㅠ
element.click()


driver.quit

<bound method ChromiumDriver.quit of <selenium.webdriver.chrome.webdriver.WebDriver (session="2bf142495ac25adfdffc7e85b871d957")>>

#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news 인 값만 찾아오시오!
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기!
- 클릭 이후 이동한 페이지에서message라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력!!

In [72]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from webdriver_manager.chrome import ChromeDriverManager
import time

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")

element01 = driver.find_element(By.NAME,"username")
# element01 = driver.find_element(By.ID,"username")
element01.clear()
element01.send_keys("error@error.com")

element02 = driver.find_element(By.NAME,"password")
# element02 = driver.find_element(By.ID,"password")
element02.clear()
element02.send_keys("1234")
#여기선 변수명 같이 사용해도 됨. 파이썬은 유연하니까~

time.sleep(3)

# element02.send_keys(Keys.RETURN) # 엔터키 입력
element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']") # CSS는 이렇게 쓰임, input 태그 많아요ㅠ
element.click()

elements = driver.find_elements(By.CSS_SELECTOR,"div.message")
for element in elements:
    print(element.text)


driver.quit

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~


<bound method ChromiumDriver.quit of <selenium.webdriver.chrome.webdriver.WebDriver (session="bd8017a4f9df37bd2c3b266608c39fed")>>

#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news 인 값만 찾아오시오!
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기!
- 클릭 이후 이동한 페이지에서message라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력!!
- 해당 페이지에서 로그아웃 버튼을 클릭 후 이동한 페이지에서 다시 message 라는 속성값을 가지고 있는 요소와 텍스트 정보를 출력!

In [79]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from webdriver_manager.chrome import ChromeDriverManager
import time

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")

element01 = driver.find_element(By.NAME,"username")
# element01 = driver.find_element(By.ID,"username")
element01.clear()
element01.send_keys("error@error.com")

element02 = driver.find_element(By.NAME,"password")
# element02 = driver.find_element(By.ID,"password")
element02.clear()
element02.send_keys("1234")
#여기선 변수명 같이 사용해도 됨. 파이썬은 유연하니까~

time.sleep(3)

# element02.send_keys(Keys.RETURN) # 엔터키 입력
element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']") # CSS는 이렇게 쓰임, input 태그 많아요ㅠ
element.click()

elements = driver.find_elements(By.CSS_SELECTOR,"div.message")
for element in elements:
    print(element.text)

time.sleep(3)

element03 = driver.find_element(By.CSS_SELECTOR,"input[type='submit']")
element03.click()

elements = driver.find_elements(By.CSS_SELECTOR,"div.message")
for element in elements:
    print(element.text)

driver.quit

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~
실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~


<bound method ChromiumDriver.quit of <selenium.webdriver.chrome.webdriver.WebDriver (session="b5bf243bcc1953c50d05c13b0772f3c0")>>